src/Faker/Provider/Lorem.php

```diff
@@ -198,6 +198,6 @@ public static function text($maxNbChars = 200)
 
     protected static function randomizeNbElements($nbElements)
     {
-        return (int) ($nbElements * mt_rand(60, 140) / 100) + 1;
+        return (int) ($nbElements * static::numberBetween(60, 140) / 100) + 1;
```

Comment:
these should all be `self::`

Reply:
What's the reason behind this? I've seen both `self::` and `static::` calls for this method in existing code. I will unify the calls in one of the ways if needed.

Reply:
`static` refers to the class the method is defined on, `self` refers to the current class. 

So if you overwrite a method, and use `static`, it won't use the overwritten implementation. 

Source: <https://stackoverflow.com/a/11710168>

Reply:
@bramceulemans I'm familiar with that - that's why I think `static` is better because it allows us to extend the provider and override `numberBetween()` method. However, I don't think anyone should override this method so not sure if it's worth it.

Reply:
I don't think we should allow people to do that. We should favour composition over extension, and probably make this class final in 2.x.

Reply:
That is, it is really intentional that the overwritten implementation is not used.

Reply:
@GrahamCampbell changed to `self::` then :wink: